### DISCUSSION TITLE
Giving up GitHub

### DISCUSSION THREAD

**Discussion by mensch72** [pik-gane/vodle#215] [General]
Because of the reasons mentioned in the footer of the [README.md](https://github.com/pik-gane/vodle#readme), we'd like to decrease our dependence on GitHub as much as possible over time. 

A possible strategy for this could be:
1. Stop making changes (both online, by pushing, or via Crowdin) to this GitHub repository and its issues and PRs
2. Label all discussions as "discussion" and convert them to issues (since Codeberg does not have discussions)
3. "Migrate" it to a fresh repository on [CodeBerg](https://codeberg.org/)
4. Manually adjust the new repo's settings, including branch protection rules
5. Manually migrate non-code files (e.g. screenshots) linked to in .md files
6. Repair links in .md files (to issues, milestones, github-hosted non-code files)
7. Start making changes only to that repository, including new issues and PRs
8. If necessary, start using the Crowdin CLI to push Crowdin translations if not otherwise possible 
9. Keep this copy on GitHub for some time as a mirror that is regularly updated *(can this be automated?)*, thereby keep profiting from GitHub services and tools not provided by the other hosting service, such as security and other code analysis, until replacements have been identified.

As a prerequisite, we should research whether codeberg...
- supports contributor license agreement requirements
- can be integrated with Crowdin
- can be integrated with any Kanban board
- can be used with projects such as GoodFirstIssue, UpForGrabs, Hacktoberfest, etc.

**Comment by mensch72** [comment]
As long as we are on GitHub, we also need to find a way that enables contributors to contribute without using GitHub if they want.

**Comment by mensch72** [comment]
# Codeberg

Just tested a "migration" to codeberg. Overall this looks very good!

## Results
- Apparently correctly migrated: repo labels,, branches (including foreign source branches of PRs!), code, commits, issues with labels and milestones, PRs with labels, releases, tags, wiki, some links in md files (code)
- **not** migrated, but could have been: issue **assignees**, **discussions**, some **links** in md files (issues, milestones, non-code files on github), non-code files on github (e.g. **screenshots**)
- rightfully not migrated: links to github users, signature keys (obviously)

## Lessons learned
- discussions should be converted to issues with label "discussion" before migrating to retain them.
- assignees of issues should be politely asked to join codeberg before migrating and to self-assign to their issues after migration (if possible) 

## Branch protection rules
They include:
- Disable push
- Enable merge whitelist
- Enable status check (by number of approvals)
- Restrict approvals to whitelisted users
- Require signed commits

## Signing settings still need to be understood:
![image](https://user-images.githubusercontent.com/22815964/199830542-6b13aa9d-0dbd-45d4-b712-7015cc8b2c87.png)

**Comment by comradekingu** [comment]
In that same vein, use Weblate instead of Crowdin?

**Comment by mensch72** [comment]


**Reply by comradekingu** [reply]
https://hosted.weblate.org/hosting/ is a nice offering.

**Reply by mensch72** [reply]
Thx for that recommendation. We're using it now.

**Comment by mensch72** [comment]
As long as Codeberg does not have Kanban boards, a good alternative for the project management things seems to be [Twake](https://twake.app/en/).
